pkg/kubectl/cmd/run.go

```diff
@@ -165,6 +165,20 @@ func RunRun(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *c
 		return fmt.Errorf("Invalid image name %q: %v", imageName, reference.ErrReferenceInvalidFormat)
 	}
 
+	outputFormat := cmdutil.GetFlagString(cmd, "output")
+	showLabelsFlag := cmdutil.GetFlagBool(cmd, "show-labels")
+	dryRunFlag := cmdutil.GetDryRunFlag(cmd)
+	if showLabelsFlag {
+		if dryRunFlag {
```

Comment:
I think you need to re-org the conditions.

`outputFormat != "" && outputFormat != "wide"` this will always be `false`.

Reply:
When `outputFormat` is `""` or `"wide"` , this will be false. But `outputFormat` could be other value, like `"yaml"`. When meet this condition, it should error out as expectation.

Reply:
@charrywanganthony I mean  `outputFormat` can only be a certain value. The condition `outputFormat != "" && outputFormat != "wide"` will be not met at the same time.

Reply:
`condition1 != condition2 && condition1 != condition3` finally it's true. I'm a little confused. :crying_cat_face:  Is this not true?

Reply:
Yeah. I got a mistake.

Reply:
Never mind. :smile: could you please add an ok-to-test label?